pkg/controller/perconaservermongodbbackup/backup.go

```diff
@@ -68,11 +69,17 @@ func (b *Backup) Start(ctx context.Context, k8sclient client.Client, cluster *ap
 		compLevel = &l
 	}
 
+	t := cr.Spec.Type
+	if t == backup.BackupTypeIncrementalBase {
+		t = defs.IncrementalBackup
+	}
+
 	cmd := ctrl.Cmd{
 		Cmd: ctrl.CmdBackup,
 		Backup: &ctrl.BackupCmd{
 			Name:             name,
-			Type:             cr.Spec.Type,
+			Type:             t,
+			IncrBase:         cr.Spec.Type == backup.BackupTypeIncrementalBase,
```

Comment:
Maybe it is worth having here 

```
	cmd := ctrl.Cmd{
		Cmd: ctrl.CmdBackup,
		Backup: &ctrl.BackupCmd{
			Name:             name,
			Type:             cr.Spec.BackupType(),
			IncrBase:         cr.Spec.IsBackupTypeIncrementalBase(),
			Compression:      cr.Spec.Compression,
			CompressionLevel: compLevel,
		},
	}
```

And having inside the spec some functions like the following:

```
func (p PerconaServerMongoDBBackupSpec) IsBackupTypeIncrementalBase() bool
func (p PerconaServerMongoDBBackupSpec) BackupType() defs.BackupType
```

Mainly because this logic belongs to the backup spec type and less to the backup and cmd creation itself.

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1836/commits/f53376062c271385d7b4d9177498ddf727ad335e